### ISSUE TITLE
xdis-3.6.0.tar.gz on pypi, tests fail

### ISSUE THREAD

**Issue by dol-sen** [rocky/python-xdis#39]
As you know I've had problems with packaging xdis for Gentoo for some time, getting test fails.
I was trying to add the 3.6.0 release today and was getting stuck on several python-2.7 test fails.
I updated my checkout of this repo, ran the tests from it.  All tests pass for all python versions we currently support.
I also noticed that there were more tests ran from the local checkout than from the install tarball downloaded from pypi.

So, I made a new tarball using setup.py sdist, re-generated our ebuild manifest for the new tarball and it runs all tests without failure.  Plus it runs more tests than did the pypi tarball.
Github archives and release also work correctly.

P.S. the only patch applied is to remove setup_requires from setup.py

**Comment by rocky** [comment]
I know you package xdis for Gentoo - many thanks! I wasn't aware that you were getting test failures, but I am glad that things are okay now.  Alas, there will be yet another big release soon - this project still needs to grow a bit. If you want to get ahead of the game, test what is in git right now. Soon it will be released.

Given the concerns you have, I've made another pass over the administrative-side setup in commit d2bf4dd . As for tests that run from the local checkout, I've included in the source distribution the Makefiles which glue the various kinds of tests together. However I think you'll need setup_requires in setup.py to run those. In general if you are having problems with the tests, look at `circle.yml` or `.travis.yml` to see how they get their tests run. 

CI testing basically tests via `make`, but here's an overview. (BTW if you use [`remake`](https://packages.gentoo.org/packages/sys-devel/remake), then `remake --tasks` will suggest the below too. )

If you run `make check`. It runs pytests as well as the integration test in the `test` directory. `make check-full` runs all of this bug also checks every python library file in the standard libary for the version of Python you have installed.

**Comment by dol-sen** [comment]
I used setup.py test for this last release.  Previous ones I ran pytest -v and test_pyenvlib.py --verify --simple --compile.

Seems I also mixed up test fails between xdis and uncompyle6.  I have had a few fails in past xdis (I missed some releases), but not like I always get in uncompyle6.  I also confirmed that running setup.py test without pytest-runner, all the same tests run as with it installed, and none fail.  Also, if it is required for development testing, shouldn't it be in test_requires?

I did run the tests with the current xdis HEAD, all pass :)
I'll test uncompyle6 from git, see if it has less/no test fails compared to the pypi tarballs.  But I think some of the errors there might utf-8 related.  Our python-2.7 is installed with uft-8 (system wide default) t which causes twisted to give us some test fails.
